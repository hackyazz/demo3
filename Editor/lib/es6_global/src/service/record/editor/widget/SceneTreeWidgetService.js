


function getWidget(param) {
  return /* SceneTree */0;
}

function isWidget(startWidget) {
  if (startWidget !== undefined) {
    return startWidget === /* SceneTree */0;
  } else {
    return false;
  }
}

export {
  getWidget ,
  isWidget ,
  
}
/* No side effect */
