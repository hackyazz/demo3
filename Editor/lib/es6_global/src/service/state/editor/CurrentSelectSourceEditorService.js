


function getCurrentSelectSource(editorState) {
  return editorState[/* currentSelectSource */12];
}

function setCurrentSelectSource(currentSelectSource, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */currentSelectSource,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function clearCurrentSelectSource(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */undefined,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getCurrentSelectSource ,
  setCurrentSelectSource ,
  clearCurrentSelectSource ,
  
}
/* No side effect */
