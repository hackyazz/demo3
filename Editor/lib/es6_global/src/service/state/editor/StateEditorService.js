

import * as CreateEditorStateDataEditorService$WonderEditor from "../data/CreateEditorStateDataEditorService.js";

function getStateIsDebug(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1];
}

function setStateIsDebug(isDebug) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1] = isDebug;
  return /* () */0;
}

function getIsRun(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2];
}

function setIsRun(isRun) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2] = isRun;
  return /* () */0;
}

function getState(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0];
}

function setState(state) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0] = state;
  return state;
}

export {
  getStateIsDebug ,
  setStateIsDebug ,
  getIsRun ,
  setIsRun ,
  getState ,
  setState ,
  
}
/* CreateEditorStateDataEditorService-WonderEditor Not a pure module */
