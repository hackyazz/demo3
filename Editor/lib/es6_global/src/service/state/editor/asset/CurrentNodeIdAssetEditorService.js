

import * as NodeAssetService$WonderEditor from "../../../record/editor/asset/NodeAssetService.js";
import * as CurrentNodeIdAssetService$WonderEditor from "../../../record/editor/asset/CurrentNodeIdAssetService.js";
import * as RootTreeAssetEditorService$WonderEditor from "./RootTreeAssetEditorService.js";

function getCurrentNodeId(editorState) {
  return CurrentNodeIdAssetService$WonderEditor.getCurrentNodeId(editorState[/* assetRecord */2]);
}

function unsafeGetCurrentNodeId(editorState) {
  return CurrentNodeIdAssetService$WonderEditor.unsafeGetCurrentNodeId(editorState[/* assetRecord */2]);
}

function clearCurrentNodeId(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */CurrentNodeIdAssetService$WonderEditor.clearCurrentNodeId(editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function setCurrentNodeId(currentNodeId, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */CurrentNodeIdAssetService$WonderEditor.setCurrentNodeId(currentNodeId, editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function couldRemoveCurrentNode(editorState) {
  var match = CurrentNodeIdAssetService$WonderEditor.getCurrentNodeId(editorState[/* assetRecord */2]);
  if (match !== undefined) {
    return !NodeAssetService$WonderEditor.isIdEqual(match, NodeAssetService$WonderEditor.getNodeId(RootTreeAssetEditorService$WonderEditor.getRootNode(editorState)));
  } else {
    return false;
  }
}

export {
  getCurrentNodeId ,
  unsafeGetCurrentNodeId ,
  clearCurrentNodeId ,
  setCurrentNodeId ,
  couldRemoveCurrentNode ,
  
}
/* CurrentNodeIdAssetService-WonderEditor Not a pure module */
