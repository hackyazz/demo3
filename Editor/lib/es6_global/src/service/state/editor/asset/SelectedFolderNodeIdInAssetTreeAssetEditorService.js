

import * as SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor from "../../../record/editor/asset/SelectedFolderNodeIdInAssetTreeAssetService.js";

function getSelectedFolderNodeIdInAssetTree(editorState) {
  return SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.getSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]);
}

function unsafeGetSelectedFolderNodeIdInAssetTree(editorState) {
  return SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.unsafeGetSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]);
}

function clearSelectedFolderNodeIdInAssetTree(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.clearSelectedFolderNodeIdInAssetTree(editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function setSelectedFolderNodeIdInAssetTree(selectedFolderNodeIdInAssetTree, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */SelectedFolderNodeIdInAssetTreeAssetService$WonderEditor.setSelectedFolderNodeIdInAssetTree(selectedFolderNodeIdInAssetTree, editorState[/* assetRecord */2]),
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */editorState[/* currentSelectSource */12],
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getSelectedFolderNodeIdInAssetTree ,
  unsafeGetSelectedFolderNodeIdInAssetTree ,
  clearSelectedFolderNodeIdInAssetTree ,
  setSelectedFolderNodeIdInAssetTree ,
  
}
/* SelectedFolderNodeIdInAssetTreeAssetService-WonderEditor Not a pure module */
