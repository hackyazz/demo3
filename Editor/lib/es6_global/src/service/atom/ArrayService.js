

import * as Curry from "../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Caml_obj from "../../../../../node_modules/bs-platform/lib/es6/caml_obj.js";
import * as Caml_option from "../../../../../node_modules/bs-platform/lib/es6/caml_option.js";
import * as Log$WonderLog from "../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as Contract$WonderLog from "../../../../../node_modules/wonder-log/lib/es6_global/src/Contract.js";
import * as Result$WonderEditor from "../../module/Result.js";
import * as OptionService$WonderEditor from "../primitive/OptionService.js";
import * as ArrayService$WonderCommonlib from "../../../../../node_modules/wonder-commonlib/lib/es6_global/src/ArrayService.js";
import * as StateEditorService$WonderEditor from "../state/editor/StateEditorService.js";
import * as MutableHashMapService$WonderCommonlib from "../../../../../node_modules/wonder-commonlib/lib/es6_global/src/MutableHashMapService.js";

function create(param) {
  return /* array */[];
}

function range(a, b) {
  var result = ArrayService$WonderCommonlib.createEmpty(/* () */0);
  for(var i = a; i <= b; ++i){
    result.push(i);
  }
  return result;
}

function unsafeGetFirst(arr) {
  return Contract$WonderLog.ensureCheck((function (r) {
                return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("first array element exist", "not"), (function (param) {
                              return Contract$WonderLog.assertNullableExist(r);
                            }));
              }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0), arr[0]);
}

function getFirst(arr) {
  return Caml_option.nullable_to_opt(arr[0]);
}

function unsafeGetLast(arr) {
  return Contract$WonderLog.ensureCheck((function (r) {
                return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("last array element exist", "not"), (function (param) {
                              return Contract$WonderLog.assertNullableExist(r);
                            }));
              }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0), arr[arr.length - 1 | 0]);
}

function getLast(arr) {
  return Caml_option.nullable_to_opt(arr[arr.length - 1 | 0]);
}

function removeLast(arr) {
  Contract$WonderLog.requireCheck((function (param) {
          return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("array length should >= 1", "not"), (function (param) {
                        return Contract$WonderLog.Operators[/* >= */7](arr.length, 1);
                      }));
        }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0));
  return OptionService$WonderEditor.unsafeGet(Caml_option.undefined_to_opt(arr.pop()));
}

function removeFirst(arr) {
  Contract$WonderLog.requireCheck((function (param) {
          return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("array length should >= 1", "not"), (function (param) {
                        return Contract$WonderLog.Operators[/* >= */7](arr.length, 1);
                      }));
        }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0));
  return OptionService$WonderEditor.unsafeGet(Caml_option.undefined_to_opt(arr.shift()));
}

function unsafeGetNth(index, arr) {
  return Contract$WonderLog.ensureCheck((function (r) {
                return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("array[" + (String(index) + "] element exist"), "not"), (function (param) {
                              return Contract$WonderLog.assertNullableExist(r);
                            }));
              }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0), arr[index]);
}

function hasItem(arr) {
  var match = arr.length > 0;
  if (match) {
    return true;
  } else {
    return false;
  }
}

function unshift(item, arr) {
  arr.unshift(item);
  return arr;
}

function push(item, arr) {
  arr.push(item);
  return arr;
}

function pushMany(itemArr, arr) {
  return ArrayService$WonderCommonlib.reduceOneParam((function (arr, item) {
                return push(item, arr);
              }), arr, itemArr);
}

function hasItemByFunc(func, arr) {
  return hasItem(arr.filter(Curry.__1(func)));
}

function removeDuplicateItems(buildKeyFunc, arr) {
  var resultArr = /* array */[];
  var map = MutableHashMapService$WonderCommonlib.createEmpty(/* () */0);
  for(var i = 0 ,i_finish = arr.length - 1 | 0; i <= i_finish; ++i){
    var item = arr[i];
    var key = buildKeyFunc(item);
    var match = MutableHashMapService$WonderCommonlib.get(key, map);
    if (match === undefined) {
      resultArr.push(item);
      MutableHashMapService$WonderCommonlib.set(key, item, map);
    }
    
  }
  return resultArr;
}

function exclude(excludeArr, arr) {
  return arr.filter((function (value) {
                return !excludeArr.includes(value);
              }));
}

function intersect(arr1, arr2) {
  return arr1.filter((function (value) {
                return arr2.includes(value);
              }));
}

function hasIntersect(arr1, arr2) {
  return intersect(arr1, arr2).length > 0;
}

function fastConcat(arr1, arr2) {
  return ArrayService$WonderCommonlib.reduceOneParam((function (arr1, value2) {
                return push(value2, arr1);
              }), arr1, arr2);
}

function fastConcatArrays(arrayArrays) {
  Contract$WonderLog.requireCheck((function (param) {
          return Contract$WonderLog.test(Log$WonderLog.buildAssertMessage("length >= 1", "not"), (function (param) {
                        return Contract$WonderLog.Operators[/* >= */7](arrayArrays.length, 1);
                      }));
        }), StateEditorService$WonderEditor.getStateIsDebug(/* () */0));
  return ArrayService$WonderCommonlib.reduceOneParam(fastConcat, arrayArrays[0], arrayArrays.slice(1));
}

function isEqual(arr1, arr2) {
  return Caml_obj.caml_equal(arr1.sort(), arr2.sort());
}

function isInclude(sourceArr, targetArr) {
  return targetArr.filter((function (value) {
                return !sourceArr.includes(value);
              })).length === 0;
}

function _addFailureFunc(param, param$1) {
  return /* tuple */[
          param[0],
          Curry._1(param[1], param$1[1])
        ];
}

function _handleFailAndSuceessFunc(param, s) {
  return /* tuple */[
          param[0],
          Curry._1(param[1], s)
        ];
}

function _handleSuceessFuncAndFailFunc(func, param) {
  return /* tuple */[
          param[0],
          Curry._1(func, param[1])
        ];
}

function traverseSameDataResultAndCollectByApply(traverseFunc, arr) {
  var func = Result$WonderEditor.SameDataResult[/* apply */2];
  var applyFunc = function (param) {
    var func$1 = Curry._1(func, param);
    return (function (param) {
        return Curry._4(func$1, param, _addFailureFunc, _handleFailAndSuceessFunc, _handleSuceessFuncAndFailFunc);
      });
  };
  var returnFunc = Result$WonderEditor.SameDataResult[/* success */0];
  return ArrayService$WonderCommonlib.reduceOneParam((function (resultArr, value) {
                return applyFunc(applyFunc(Curry._1(returnFunc, push))(Curry._1(traverseFunc, value)))(resultArr);
              }), Curry._1(returnFunc, /* array */[]), arr);
}

export {
  create ,
  range ,
  unsafeGetFirst ,
  getFirst ,
  unsafeGetLast ,
  getLast ,
  removeLast ,
  removeFirst ,
  unsafeGetNth ,
  hasItem ,
  unshift ,
  push ,
  pushMany ,
  hasItemByFunc ,
  removeDuplicateItems ,
  exclude ,
  intersect ,
  hasIntersect ,
  fastConcat ,
  fastConcatArrays ,
  isEqual ,
  isInclude ,
  _addFailureFunc ,
  _handleFailAndSuceessFunc ,
  _handleSuceessFuncAndFailFunc ,
  traverseSameDataResultAndCollectByApply ,
  
}
/* Log-WonderLog Not a pure module */
