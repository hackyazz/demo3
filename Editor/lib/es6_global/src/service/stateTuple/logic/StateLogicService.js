

import * as Curry from "../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as StateEditorService$WonderEditor from "../../state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../state/engine/state/StateEngineService.js";
import * as DirectorEngineService$WonderEditor from "../../state/engine/DirectorEngineService.js";

function getEngineStateToGetData(handleFunc) {
  return Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0));
}

function getAndSetEngineState(handleFunc) {
  StateEngineService$WonderEditor.setState(Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function refreshEngineState(engineState) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, engineState));
  return /* () */0;
}

function refreshEngineStateAndReturnEngineState(engineState) {
  return DirectorEngineService$WonderEditor.loopBody(0, engineState);
}

function getAndRefreshEngineState(param) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function getAndRefreshEngineStateWithFunc(handleFunc) {
  StateEngineService$WonderEditor.setState(DirectorEngineService$WonderEditor.loopBody(0, Curry._1(handleFunc, StateEngineService$WonderEditor.unsafeGetState(/* () */0))));
  return /* () */0;
}

function loopBodyWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getAndRefreshEngineStateWhenStop(param) {
  StateEngineService$WonderEditor.setState(loopBodyWhenStop(StateEngineService$WonderEditor.unsafeGetState(/* () */0)));
  return /* () */0;
}

function renderWhenStop(engineState) {
  var match = StateEditorService$WonderEditor.getIsRun(/* () */0);
  if (match) {
    return engineState;
  } else {
    return DirectorEngineService$WonderEditor.loopBody(0, engineState);
  }
}

function getEditorState(handleFunc) {
  return Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0));
}

function getAndSetEditorState(handleFunc) {
  StateEditorService$WonderEditor.setState(Curry._1(handleFunc, StateEditorService$WonderEditor.getState(/* () */0)));
  return /* () */0;
}

function getStateToGetData(handleFunc) {
  return Curry._1(handleFunc, /* tuple */[
              StateEditorService$WonderEditor.getState(/* () */0),
              StateEngineService$WonderEditor.unsafeGetState(/* () */0)
            ]);
}

function getAndSetState(handleFunc) {
  var match = Curry._1(handleFunc, /* tuple */[
        StateEditorService$WonderEditor.getState(/* () */0),
        StateEngineService$WonderEditor.unsafeGetState(/* () */0)
      ]);
  StateEditorService$WonderEditor.setState(match[0]);
  StateEngineService$WonderEditor.setState(match[1]);
  return /* () */0;
}

function setState(param) {
  StateEditorService$WonderEditor.setState(param[0]);
  StateEngineService$WonderEditor.setState(param[1]);
  return /* () */0;
}

export {
  getEngineStateToGetData ,
  getAndSetEngineState ,
  refreshEngineState ,
  refreshEngineStateAndReturnEngineState ,
  getAndRefreshEngineState ,
  getAndRefreshEngineStateWithFunc ,
  loopBodyWhenStop ,
  getAndRefreshEngineStateWhenStop ,
  renderWhenStop ,
  getEditorState ,
  getAndSetEditorState ,
  getStateToGetData ,
  getAndSetState ,
  setState ,
  
}
/* StateEditorService-WonderEditor Not a pure module */
