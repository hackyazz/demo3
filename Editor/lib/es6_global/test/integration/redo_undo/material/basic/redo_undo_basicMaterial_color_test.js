

import * as Curry from "../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as ArrayService$WonderEditor from "../../../../../src/service/atom/ArrayService.js";
import * as RedoUndoTool$WonderEditor from "../../tool/RedoUndoTool.js";
import * as PickColorTool$WonderEditor from "../../../../tool/PickColorTool.js";
import * as GameObjectTool$WonderEditor from "../../../../tool/GameObjectTool.js";
import * as BuildCanvasTool$WonderEditor from "../../../../tool/BuildCanvasTool.js";
import * as EventListenerTool$WonderEditor from "../../../../unit/tool/EventListenerTool.js";
import * as StateLogicService$WonderEditor from "../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as StateEditorService$WonderEditor from "../../../../../src/service/state/editor/StateEditorService.js";
import * as MainEditorAssetTool$WonderEditor from "../../../asset/tool/MainEditorAssetTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../tool/MainEditorSceneTool.js";
import * as SceneTreeWidgetService$WonderEditor from "../../../../../src/service/record/editor/widget/SceneTreeWidgetService.js";
import * as BuildComponentForCurryTool$WonderEditor from "../../../../tool/BuildComponentForCurryTool.js";
import * as MainEditorBasicMaterialTool$WonderEditor from "../../../inspector/composable_component/sceneTree_inspector/renderGroup/material/tool/MainEditorBasicMaterialTool.js";
import * as CurrentNodeIdAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/CurrentNodeIdAssetEditorService.js";
import * as CurrentSelectSourceEditorService$WonderEditor from "../../../../../src/service/state/editor/CurrentSelectSourceEditorService.js";
import * as SelectedFolderNodeIdInAssetTreeAssetEditorService$WonderEditor from "../../../../../src/service/state/editor/asset/SelectedFolderNodeIdInAssetTreeAssetEditorService.js";

describe("redo_undo: basicMaterial color", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        var _changeColorAndPushUndoStack = function (materialComponent, color) {
          var sourceColor = MainEditorBasicMaterialTool$WonderEditor.getColor(materialComponent);
          MainEditorBasicMaterialTool$WonderEditor.changeColor(materialComponent, color);
          return MainEditorBasicMaterialTool$WonderEditor.closeColorPicker(materialComponent, sourceColor, undefined, undefined, /* () */0);
        };
        var _simulateTwiceChangeColor = function () {
          BuildCanvasTool$WonderEditor.buildFakeCanvas(sandbox);
          var currentGameObjectMaterial = GameObjectTool$WonderEditor.getCurrentGameObjectBasicMaterial(/* () */0);
          var color1 = PickColorTool$WonderEditor.buildColor1(/* () */0);
          var color2 = PickColorTool$WonderEditor.buildColor2(/* () */0);
          _changeColorAndPushUndoStack(currentGameObjectMaterial, color1);
          return _changeColorAndPushUndoStack(currentGameObjectMaterial, color2);
        };
        var _beforeEach = function () {
          MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, (function () {
                  MainEditorAssetTool$WonderEditor.initAssetTree(/* () */0);
                  return MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode(/* () */0);
                }));
          var partial_arg = SceneTreeWidgetService$WonderEditor.getWidget(/* () */0);
          StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                  return CurrentSelectSourceEditorService$WonderEditor.setCurrentSelectSource(partial_arg, param);
                }));
          return MainEditorBasicMaterialTool$WonderEditor.changeMaterialTypeToBeBasicMaterial(undefined, undefined, /* () */0);
        };
        var _afterEach = function () {
          StateEditorService$WonderEditor.setState(SelectedFolderNodeIdInAssetTreeAssetEditorService$WonderEditor.clearSelectedFolderNodeIdInAssetTree(CurrentNodeIdAssetEditorService$WonderEditor.clearCurrentNodeId(StateEditorService$WonderEditor.getState(/* () */0))));
          return /* () */0;
        };
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                return Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        return RedoUndoTool$WonderEditor.testRedoUndoTwoStep(sandbox, "prepare first step: set currentSceneTreeNode", /* tuple */[
                    _simulateTwiceChangeColor,
                    _beforeEach,
                    _afterEach
                  ], BuildComponentForCurryTool$WonderEditor.buildBasicMaterial);
      }));

export {
  
}
/*  Not a pure module */
