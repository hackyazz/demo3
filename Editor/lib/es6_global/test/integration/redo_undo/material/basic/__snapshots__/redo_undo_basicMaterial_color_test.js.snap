// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redo_undo: basicMaterial color prepare first step: set currentSceneTreeNode test redo operate test redo one step undo step which from second to zero, redo step which from zero to first 1`] = `
<article
  className="wonder-basic-material"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#7df1e8",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
</article>
`;

exports[`redo_undo: basicMaterial color prepare first step: set currentSceneTreeNode test redo operate test redo two step undo step which from second to zero,redo step which from zero to second 1`] = `
<article
  className="wonder-basic-material"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#1918e8",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
</article>
`;

exports[`redo_undo: basicMaterial color prepare first step: set currentSceneTreeNode test undo operate test not undo 1`] = `
<article
  className="wonder-basic-material"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#1918e8",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
</article>
`;

exports[`redo_undo: basicMaterial color prepare first step: set currentSceneTreeNode test undo operate test undo one step step which from second to first 1`] = `
<article
  className="wonder-basic-material"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#7df1e8",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
</article>
`;

exports[`redo_undo: basicMaterial color prepare first step: set currentSceneTreeNode test undo operate test undo two step step which from second to zero 1`] = `
<article
  className="wonder-basic-material"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Color
    </div>
    <div
      className="item-content item-color"
    >
      <div
        className="color-hex"
        onClick={[Function]}
        style={
          Object {
            "background": "#ffffff",
          }
        }
      />
      <div
        className="color-select"
        onClick={[Function]}
      >
        <img
          src="./public/img/color.png"
        />
      </div>
    </div>
  </article>
</article>
`;
